src/ImageSharp.Drawing/Processing/Processors/Drawing/DrawImageProcessor.cs

```diff
@@ -76,23 +78,18 @@ protected override void OnFrameApply(ImageFrame<TPixel> source, Rectangle source
 
             int width = maxX - minX;
 
-            MemoryAllocator memoryAllocator = this.Image.GetConfiguration().MemoryAllocator;
-
-            using (IMemoryOwner<float> amount = memoryAllocator.Allocate<float>(width))
-            {
-                amount.GetSpan().Fill(this.Opacity);
-
-                ParallelFor.WithConfiguration(
-                    minY,
-                    maxY,
-                    configuration,
-                    y =>
-                        {
-                            Span<TPixel> background = source.GetPixelRowSpan(y).Slice(minX, width);
-                            Span<TPixel> foreground = targetImage.GetPixelRowSpan(y - locationY).Slice(targetX, width);
-                            blender.Blend(memoryAllocator, background, background, foreground, amount.GetSpan());
-                        });
-            }
+            MemoryAllocator memoryAllocator = this.Image.GetConfiguration().MemoryAllocator;
+
+            ParallelFor.WithConfiguration(
+                minY,
+                maxY,
+                configuration,
+                y =>
+                    {
+                        Span<TPixelDst> background = source.GetPixelRowSpan(y).Slice(minX, width);
+                        Span<TPixelSrc> foreground = targetImage.GetPixelRowSpan(y - locationY).Slice(targetX, width);
+                        blender.Blend<TPixelSrc>(memoryAllocator, background, background, foreground, this.Opacity);
```

Comment:
I couldn't find the callee `Blend` method in the diff but it's imperative now that if we are mixing pixel formats and conversion to/from Vector4 needs to used the scaled conversion methods.